src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceTransformer.java

```diff
@@ -635,6 +635,15 @@ private StaticLangRpcStubView generateRpcStubClass(
     stubClass.hasDefaultInstance(interfaceConfig.hasDefaultInstance());
     stubClass.hasLongRunningOperations(interfaceConfig.hasLongRunningOperations());
 
+    stubClass.transportProtocol(productConfig.getTransportProtocol());
+    if (productConfig.getTransportProtocol() == TransportProtocol.HTTP) {
```

Comment:
nit: This is fine, but I'd consider a look-up table here.

Reply:
This is how these types of differences were reflected in the other parts of gapic-generator (i.e. if statements like this are pretty common). Not very pretty, but given there are only two options, lookup table seems not necessary.

Reply:
sgtm